crates/uv/tests/branching_urls.rs

```diff
@@ -603,7 +603,7 @@ fn branching_urls_of_different_sources_disjoint() -> Result<()> {
     [[distribution]]
     name = "iniconfig"
     version = "2.0.0"
-    source = { git = "https://github.com/pytest-dev/iniconfig?rev=93f5930e668c0d1ddf4597e38dd0dea4e2665e7a#93f5930e668c0d1ddf4597e38dd0dea4e2665e7a" }
+    source = { git = "https://github.com/pytest-dev/iniconfig", precise = "93f5930e668c0d1ddf4597e38dd0dea4e2665e7a", revision = "93f5930e668c0d1ddf4597e38dd0dea4e2665e7a" }
```

Comment:
Should we omit the revision if they _are_ the same?

Reply:
So I am in favor of this, but I think it then becomes ambiguous. Specifically, I'm thinking about this part in the code:

https://github.com/astral-sh/uv/blob/65cbeb9fe670d829ef3fd92a437bd75d4d4968bf/crates/uv-resolver/src/lock.rs#L1372-L1396

Namely, this is where we deserialize our structured git data back into a `Source`. Currently, when `branch`, `tag` and `revision` are all missing, we assume "default branch" semantics. Which means that if we drop `revision = "..."` when it's identical to `commit = "..."`, we won't be able to distinguish between "user specified a specific revision" and "user asked for the default branch."